src/instrumenter/instrumentation_context.ts

```diff
@@ -467,6 +471,52 @@ export class InstrumentationContext {
         }
     }
 
+    addStringLiteralToAdjust(lit: Literal, loc: ASTNode): void {
+        this.litAdjustMap.set(lit, loc);
+    }
+
+    adjustStringLiterals(
+        contents: string,
+        units: SourceUnit,
+        fileInd: number,
+        srcMap: SrcRangeMap
+    ): string {
+        for (const lit of units.getChildrenByType(Literal)) {
+            if (!(lit.kind === LiteralKind.String && this.litAdjustMap.has(lit))) {
+                continue;
+            }
+
+            const targetNode = this.litAdjustMap.get(lit) as ASTNode;
+            const newLoc = srcMap.get(targetNode);
+            const strLoc = srcMap.get(lit);
+
+            assert(newLoc !== undefined, `Missing loc for {0}`, targetNode);
+            assert(strLoc !== undefined, `Missing loc for {0}`, lit);
+            assert(
+                contents[strLoc[0]] === "'" || contents[strLoc[0]] === '"',
+                `Expected string start at {0} not {1}`,
+                strLoc[0],
+                contents[strLoc[0]]
+            );
+            assert(
+                contents.startsWith("000000:0000:000", strLoc[0] + 1),
+                `Expected 00000:000:00 at {0} not {1}`,
+                strLoc[0] + 1,
+                contents.slice(strLoc[0] + 1, strLoc[0] + 16)
+            );
+
+            contents =
+                contents.slice(0, strLoc[0] + 1) +
+                `${String(newLoc[0]).padStart(6, "0")}:${String(newLoc[1]).padStart(
+                    4,
+                    "0"
+                )}:${String(fileInd).padStart(3, "0")}` +
+                contents.slice(strLoc[0] + 16);
```

Comment:
This looks slightly complex to read: a lot of formatting is happening in-place. Would it be possible to compose source location in variable before updating `contents`? It would simplify update assignment.

Reply:
Fixed in [029c9c3](https://github.com/ConsenSys/scribble/pull/211/commits/029c9c3117655fdbc0f0f348bcd11e354e5ce2ad)